### PR TITLE
[openthermgateway] fix automatic reconnect after connection attempt fails

### LINKED ISSUE #9903: [openthermgateway] binding doesn't automatically reconnect if connection with the gateway is lost
More details about this issue can be found in the [OpenTherm Gateway binding](https://community.openhab.org/t/39160) topic on the openHAB community website.

The issue is first mentioned by Rakoon in post [#190](https://community.openhab.org/t/39160/190). Other users have reported the same issue. Several logs have been posted for analysis.

I am currently working on a solution and will create a PR after testing a snapshot version.

### PR DESCRIPTION
**Pull request by ArjenKorevaar**
Fixes #9903

Signed-off-by: Arjen Korevaar <a.korevaar@mephix.com>

### CODE REVIEW

File: bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java [hunk 1]
```diff
@@ -151,11 +147,11 @@ public void disconnected() {
...
-        if (conf != null && !explicitDisconnect && conf.connectionRetryInterval > 0) {
+        if (!explicitDisconnect && conf != null && conf.connectionRetryInterval > 0) {
+            logger.debug("Scheduling to reconnect in {} seconds.", conf.connectionRetryInterval);
+
             scheduler.schedule(() -> {
-                if (conn != null && !connecting && !conn.isConnected()) {
-                    connect();
-                }
+                connect();
             }, conf.connectionRetryInterval, TimeUnit.SECONDS);
```
Comment (J-N-K) [inline bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java:155]:
```
scheduler.schedule(this::connect, conf.connectionRetryInterval, TimeUnit.SECONDS);
```

File: bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java [hunk 2]
```diff
@@ -218,40 +214,30 @@ public void dispose() {
...
-        }
+        explicitDisconnect = false;
 
-        return connect();
-    }
-
-    private boolean connect() {
-        @Nullable
-        OpenThermGatewayConfiguration conf = config;
+        if (connecting) {
```
Comment (J-N-K) [inline bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java:223]: Where is this set?

**Review by J-N-K — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java (hunk 2 above)
Comment (ArjenKorevaar) [reply bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java:223]: It's set to true by the connecting() callback function that is invoked by OpenThermGatewaySocketConnector during run(). When a connection is established, it's then set to false by the connected() callback function.

File: bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java (hunk 1 above)
Comment (ArjenKorevaar) [reply bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java]: Changed

**Comment by ArjenKorevaar** [comment]
I am not sure why the build fails. Rebased onto 3.0.x in an attempt to fix it, but that didnt help.

If anyone could look into this?

**Comment by cpmeister** [comment]
Has this bug been fixed in the main branch yet?

**Comment by ArjenKorevaar** [comment]
@cpmeister shouldn't be.. I forked the repo, created a branch from 3.0.x, pushed the commits en pull requested it back into openhab branch 3.0.x.

**Comment by kaikreuzer** [comment]
@ArjenKorevaar PRs are only accepted against the `main` branch. If it is a critical fix, maintainers can then backport it manually to the `3.0.x` branch, but that's at the discretion of the maintainers.

Please open a new PR with this fix against `main` - thank you!

**Pull request closed without merging**